mcs/class/System.Web.Extensions/System.Web.Handlers/ScriptModule.cs

```diff
@@ -74,7 +74,7 @@ public class ScriptModule : IHttpModule
 			if (currentHandler == null)
 				return;
 			Type pageType = currentHandler.GetType ();
-			if (typeof (Page).IsAssignableFrom (pageType) && !String.IsNullOrEmpty (contentType) && contentType.StartsWith ("application/json", StringComparison.OrdinalIgnoreCase)) {
+			if (typeof (Page).IsAssignableFrom (pageType) && !String.IsNullOrEmpty (request.PathInfo) && contentType.StartsWith ("application/json", StringComparison.OrdinalIgnoreCase)) {

```

Comment:
What if the contentType is null?

Reply:
I think it's impossible! in HttpRequest.ContentType has been checked.

Reply:
https://github.com/mono/mono/blob/master/mcs/class/System.Web/System.Web/HttpRequest.cs#L419